mantle/kola/tests/rhcos/luks.go

```diff
@@ -35,6 +38,84 @@ func init() {
 			}
 		}`),
 	})
+	register.RegisterTest(&register.Test{
+		Run:                  luksTangTest,
+		ClusterSize:          0,
+		Name:                 `rhcos.luks.tang`,
+		Flags:                []register.Flag{},
+		Distros:              []string{"rhcos"},
+		Platforms:            []string{"qemu-unpriv"},
+		ExcludeArchitectures: []string{"s390x", "ppc64le"}, // no TPM support for s390x, ppc64le in qemu
+	})
+	register.RegisterTest(&register.Test{
+		Run:                  luksSSST1Test,
+		ClusterSize:          0,
+		Name:                 `rhcos.luks.sss.t1`,
+		Flags:                []register.Flag{},
+		Distros:              []string{"rhcos"},
+		Platforms:            []string{"qemu-unpriv"},
+		ExcludeArchitectures: []string{"s390x", "ppc64le"}, // no TPM support for s390x, ppc64le in qemu
+	})
+	register.RegisterTest(&register.Test{
+		Run:                  luksSSST2Test,
+		ClusterSize:          0,
+		Name:                 `rhcos.luks.sss.t2`,
+		Flags:                []register.Flag{},
+		Distros:              []string{"rhcos"},
+		Platforms:            []string{"qemu-unpriv"},
+		ExcludeArchitectures: []string{"s390x", "ppc64le"}, // no TPM support for s390x, ppc64le in qemu
+	})
+}
+
+func getEncodedTangPin(c cluster.TestCluster) string {
+	return b64Encode(getTangPin(c))
+}
+
+func startTang(c cluster.TestCluster) {
+	if err := exec.Command("sudo", "xinetd").Run(); err != nil {
+		c.Fatalf("Unable to start Tang: %v ", err)
+	}
+}
+
+func stopTang(c cluster.TestCluster) {
+	if err := exec.Command("sudo", "killall", "xinetd").Run(); err != nil {
```

Comment:
If we try to do more tests that work this way (mutate the state of the container) then we really start to *hard require* running code out of a container image because you really don't want to accidentally do this type of stuff on your host system...

Maybe we should have this test check for `os.LookupEnv("container") == "oci"` or so?

Reply:
:man_facepalming: I didn't even think of it running outside the context of a container.  What do you think of keeping track of the pids and just killing those?

Reply:
The update commit runs xinetd unprivileged now and it keeps track of the pid that is started so we don't nuke all instances of xinetd that could possibly be running.